Select model feature in process_density_map script

kvr2007 opened the pull request:
Implements an improved map/model comparison and also provides an easy way to select a good-fitting single model from a trajectory (for downstream model building). A new dependency is added in this case: `gemmi`, which is a robust (and fast) library to handle PDB and MRC files (see https://gemmi.readthedocs.io/en/latest/)

In addition, the cryofold2 tutorial code is converted into a simple command-line script `cryofold2_setup`. This way the user can try the workflow on any data.

To test run the following commands with the CryoFold2 tutorial data (https://github.com/maccallumlab/meld/tree/master/docs/tutorial/cryofold_tutorial)

```
cryofold2_setup 4ake_t_s2.mrc 1ake_s.pdb
launch_remd --console-log --platform CUDA # this command may differ if running on a cluster
process_density_map select_model 4ake_t_s2.mrc Data/trajectory.pdb --resolution 4
```

The outputs are written to a new subfolder `Postprocess`:

`Postprocess/best_model.pdb`: Auto-selected model

`Postprocess/plot/rmsd.png`: RMSD to the starting model during the simulation; use to judge convergence

`Postprocess/plot/cc_ifsc.png`: Real-space CC and iFSC between the experimental map and MD trajectory models; use to select the best model

`Postprocess/plot/fsc_step_???.png`:  Map/model FSC plots for corresponding MD steps

jlmaccal commented:
This looks cool, but I'm a bit leery of the gemmi dependency. How easy is it to install? Is it packaged through conda, etc? It would be great if we can figure out how to make it an optional dependency that's only needed for cryoEM.

kvr2007 commented:
On 06/06/2023 22:53, Justin MacCallum wrote:
>
> This looks cool, but I'm a bit leery of the gemmi dependency. How easy 
> is it to install? Is it packaged through conda, etc? It would be great 
> if we can figure out how to make it an optional dependency that's only 
> needed for cryoEM.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/maccallumlab/meld/pull/150#issuecomment-1579431980>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ADR3YEW6TDXOBR2AVSZETJTXJ6KEVANCNFSM6AAAAAAY4R4JM4>.
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>
>
gemmi is a python package (wrapping some C++ code). It can be easily 
installed via pip or conda, so this should not be a problem.

Any other tool that can use atomic scattering factors to simulate an 
electron density map from a model will be more challenging to install 
(e.g. phenix or Refmac). But I think it is a very important feature to 
have at hand compared to simulating EM maps from models using plain 
gaussians.

But I agree, it is easier to keep it as an optional dependency (like 
mdtraj or mrcfile, I guess). For instance, gemmi can be imported only 
when the script "process_density_map" is run with mode "select_model". 
Shall I edit my pull request accordingly?

jlmaccal commented:
I missed that there were conda-forge binaries. Can you add those to the packaging requirements and then we'll test that everything works.

kvr2007 commented:
Sure, but where do I find the packaging requirements file? Is it somewhere in the CI section? Or should I just place the requirement in `meld/setup.py`?

jlmaccal commented:
Well, the first place would be in the CI/testing configuration. Have a look at `build-ubuntu-latest.yml`. Once that is working, then we would need to update the packing scripts: https://github.com/conda-forge/meld-feedstock.

kvr2007 commented:
OK, just updated `build-ubuntu-latest.yml` in my branch. I also added other python dependencies that were necessary to run the script `process_density_map`. I hope did everything correctly, never really done CI myself :)

It would be great to have the newest version of meld available in conda-forge, installation from source was not trivial for me.

kvr2007 commented:
@jlmaccal , apologies for bugging you, just wanted to check on the status of this PR, is there anything else that prevents it from getting merged?

jlmaccal commented:
Just some merge conflicts in process_density_map. Can you resolve them?

kvr2007 commented:
@jlmaccal , thanks for a swift response! I've just fixed the merge conflicts
